app/lib/shared/bcpLanguageTagChecker.js

```diff
@@ -53,7 +57,11 @@ export default function (tag) {
         (s) =>
           s.subtag.toLowerCase() === variant.toLowerCase() &&
           s.type === 'variant' &&
-          s.prefix === parsed.langtag.language.language?.toLowerCase()
+          s.prefix.includes(
+            /** @type {string} */ (
+              parsed.langtag.language.language?.toLowerCase()
+            )
+          )
```

Comment:
Checking the language is not enough as a prefix might contain other parts. Valid is also:
```
de-CH-1996
pt-BR-abl1943
ja-Latn-hepburn-heploc
sl-rozaj-1994
sl-rozaj-solba-1994
sl-rozaj-bsike-1994
```

Please add them to the tests.

Here is an idea I came up with: Test against the start of the original tag. Feel free to edit that if there is a more elegant way...

```suggestion
          s.prefix.some((p) =>
            tag
              .substring(
                 0,
                 tag.toLowerCase().indexOf(variant.toLowerCase()) - 1
               )
               .toLowerCase()
               .startsWith(p.toLowerCase())
          )
```